### PR TITLE
flowlord: Add files support

### PR DESCRIPTION
**Pull request by jbsmith7741**
- Add a new files topic for flowlord to consume
- Update info to show files and their flow 
- Bug Fix for flowlord Info to properly show multi-children flows
- [x] Update Readme to show used of files 
- [x] Update tests for changes 
- [x] Run local tests on system

### CODE REVIEW

**Comment by jbsmith7741** [comment]
Should we support `**` multi-layer folder matching. current implementation which matches files does not which means deep nested packaged can be hard to read for example  `./**/*.txt` instead of `./*/*/*/*/*/*/*.txt`

**Comment by zJeremiah** [comment]
So, @jbsmith7741 , the flowlord will be reading on the files topic? I thought we were going to use flowlord instead of the files topic, but maybe it cannot  be avoided.

File: apps/taskmasters/flowlord/README.md [hunk 1]
```diff
@@ -72,6 +73,16 @@ rule = "cron=0 0 1 * * *&Offset=-6"
 template = "?date={yyyy}-{mm}-{dd}T{hh}"
 ```
 
+### files 
+schedule a task after a specified file is written. This should be used with the filewatcher taskmaster or GCP file watching service. File matching is done using the filepath.Match which does not support `**` matching. Flowlord will attempt to pull the timestampe from the filepath which will can be used in the template with the time templates `{yyyy}|{dd}|{mm}|{hh}` also the matching file will be available through the metadata of the task `{meta:file}`
```
Comment (zJeremiah) [inline apps/taskmasters/flowlord/README.md:77]: Do you mean `which can be used in the template` not `which will can be used in the template`

File: apps/taskmasters/flowlord/files.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+	json.Unmarshal(b, &e)
+	if e.ID != "" && e.Path != "" { // this is a valid gcsEvent
+		return e.Stat()
+	}
+
+	json.Unmarshal(b, &sts)
+	return sts
+}
+
+// matchFile checks the sts file Path will all file Rules that are registered with flowlord
```
Comment (zJeremiah) [inline apps/taskmasters/flowlord/files.go:92]: not sure what `Path will all file Rules that are registered` means?

**Review by zJeremiah — APPROVED** [review]
This looks fine to me, I've added a couple of comments on your comments, just to fix some grammer (and a little better understanding).

**Pull request merged**